x-pack/plugins/apm/public/hooks/use_fetcher.tsx

```diff
@@ -129,7 +134,10 @@ export function useFetcher<TReturn>(
 
       setResult((prevResult) => ({
         data: preservePreviousData ? prevResult.data : undefined, // preserve data from previous state while loading next state
-        status: FETCH_STATUS.LOADING,
+        status:
+          prevResult.status !== FETCH_STATUS.ABORTED
+            ? FETCH_STATUS.LOADING
+            : prevResult.status,
```

Comment:
I don't think it should be possible to end here if the request was aborted. Instead I suggest you returning early in the first `useEffect`:

```ts
  useEffect(() => {
    if(result.status === FETCH_STATUS.ABORTED) {
      return;
    }

    let controller: AbortController = new AbortController();
```

This way we'll never attempt to call `fn` if a fetch was aborted.

Reply:
I tried to do this but it always passed through this piece of code and the status is changed from `FETCH_STATUS.ABORTED` to `FETCH_STATUS.LOADING`

Reply:
Can you try returning after
```
const promise = fn(createAutoAbortedAPMClient(signal));
```
?

Reply:
Same behaviour, status keep changing from  `FETCH_STATUS.ABORTED` to `FETCH_STATUS.LOADING`.